sycl/include/CL/__spirv/spirv_ops.hpp

```diff
@@ -136,100 +160,372 @@ __SPIRV_ATOMICS(__SPIRV_ATOMIC_UNSIGNED, unsigned long long)
+extern SYCL_EXTERNAL void
+__spirv_SubgroupBlockWriteINTEL(__attribute__((opencl_global)) uint16_t *Ptr,
+                                dataT Data) noexcept;
 
 template <typename dataT>
-extern dataT
+extern SYCL_EXTERNAL dataT
 __spirv_SubgroupBlockReadINTEL(const __attribute__((opencl_global)) uint32_t *Ptr) noexcept;
 
 template <typename dataT>
-extern void __spirv_SubgroupBlockWriteINTEL(__attribute__((opencl_global)) uint32_t *Ptr,
-                                            dataT Data) noexcept;
+extern SYCL_EXTERNAL void
+__spirv_SubgroupBlockWriteINTEL(__attribute__((opencl_global)) uint32_t *Ptr,
+                                dataT Data) noexcept;
 
 template <typename dataT>
-extern int32_t __spirv_ReadPipe(RPipeTy<dataT> Pipe, dataT *Data,
-                                int32_t Size, int32_t Alignment) noexcept;
+extern SYCL_EXTERNAL int32_t
+__spirv_ReadPipe(RPipeTy<dataT> Pipe, dataT *Data, int32_t Size,
+                 int32_t Alignment) noexcept;
 template <typename dataT>
-extern int32_t __spirv_WritePipe(WPipeTy<dataT> Pipe, const dataT *Data,
-                                 int32_t Size, int32_t Alignment) noexcept;
+extern SYCL_EXTERNAL int32_t
+__spirv_WritePipe(WPipeTy<dataT> Pipe, const dataT *Data, int32_t Size,
+                  int32_t Alignment) noexcept;
 template <typename dataT>
-extern void __spirv_ReadPipeBlockingINTEL(RPipeTy<dataT> Pipe, dataT *Data,
-                                          int32_t Size,
-                                          int32_t Alignment) noexcept;
+extern SYCL_EXTERNAL void
+__spirv_ReadPipeBlockingINTEL(RPipeTy<dataT> Pipe, dataT *Data, int32_t Size,
+                              int32_t Alignment) noexcept;
 template <typename dataT>
-extern void __spirv_WritePipeBlockingINTEL(WPipeTy<dataT> Pipe,
-                                           const dataT *Data, int32_t Size,
-                                           int32_t Alignment) noexcept;
+extern SYCL_EXTERNAL void
+__spirv_WritePipeBlockingINTEL(WPipeTy<dataT> Pipe, const dataT *Data,
+                               int32_t Size, int32_t Alignment) noexcept;
 template <typename dataT>
-extern RPipeTy<dataT> __spirv_CreatePipeFromPipeStorage_read(
+extern SYCL_EXTERNAL RPipeTy<dataT> __spirv_CreatePipeFromPipeStorage_read(
     const ConstantPipeStorage *Storage) noexcept;
 template <typename dataT>
-extern WPipeTy<dataT> __spirv_CreatePipeFromPipeStorage_write(
+extern SYCL_EXTERNAL WPipeTy<dataT> __spirv_CreatePipeFromPipeStorage_write(
     const ConstantPipeStorage *Storage) noexcept;
 
-extern void __spirv_ocl_prefetch(const __attribute__((opencl_global)) char *Ptr,
-                                 size_t NumBytes) noexcept;
+extern SYCL_EXTERNAL void
+__spirv_ocl_prefetch(const __attribute__((opencl_global)) char *Ptr,
+                     size_t NumBytes) noexcept;
+
+SYCL_EXTERNAL
+int __spirv_ocl_printf(const __attribute__((opencl_constant)) char *fmt, ...);
```

Comment:
I don't recall exact reason we added `extern` keyword to `__spirv_ocl_prefetch` function declaration, but all declaration must be aligned.

Reply:
> Did you try to call this function from multiple TUs and link them?

Yes, multiple TUs link smoothly.

Anyway, I've aligned the declaration.